node/src/accountant/db_access_objects/receivable_dao.rs

```diff
@@ -724,92 +830,349 @@ mod tests {
+            account_status,
+            Some(ReceivableAccount {
+                wallet: first_wallet,
+                balance_wei: first_initial_balance as i128,
+                last_received_timestamp: first_previous_timestamp,
+            })
+        );
+        let log_handler = TestLogHandler::new();
+        log_handler.exists_no_log_containing(&format!("INFO: {test_name}: "));
+        log_handler.exists_log_containing(&format!(
+            "ERROR: {test_name}: Payment reception failed, rolling back:\n\
             Block #    Wallet                                     Amount            \n\
-            1234567890 0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa 123456789123456789\n\
-            2345678901 0xbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb 234567891234567891\n\
-            3456789012 0xcccccccccccccccccccccccccccccccccccccccc 345678912345678912\n\
-            TOTAL                                                 703703592703703592"
+            123456     0x0000000000000000000000000000000000616263 45678             \n\
+            789123     0x0000000000000000000000000000000000646566 111222            \n\
+            TOTAL                                                 156900"
         ));
+        // The test framework used in here, the TransactionWrapperMock, is really a smart tool. It
+        // allows  to simulate errors thrown from Statements that come from our watched transaction
+        // acting a main role in the test. That goes along with the standard purposes of mocks. One
+        // of the extra features, though, is a simulation of how a transaction would behave in
+        // reality, especially concerning the commit time or a drop of the object, not having
+        // involved itself in an explicit call of 'commit', there a situation that should leave the
+        // database unmodified (using the roll-back strategy).
+        // Why is it important to think about? You should know that the mocked wrapper based on having
+        // its own transaction that can be used for any number of initial calls done on this single
+        // piece of it, if not also interacting with a real transaction, would all be persistently
+        // written into the database. In order to be faithful to the various distinct operations
+        // that take place in our code, we do care really a lot about the terminal stage of the
+        // transaction. If we don't make it to the 'commit' call for an issue having stepped in before,
+        // the staged changes will not engrave into the database. Which is exactly you can watch in
+        // this test.
+        //
+        // Another proof about the correct course of events is that even though we hadn't prepared
+        // a result for the mocked 'commit' method, the test didn't panic because of that. Besides
+        // that, if it had been committing each transaction separately we would've come across at
+        // least one call of 'commit' (and crashed), yet we didn't.
```

Comment:
I've rephrased this comment so that I can understand `TransactionWrapperMock` in a better way. If you'd like you can use the rephrased words in the comment. :pray:

If I've confused something, please correct me.

```rust
// This test uses the TransactionWrapperMock, a powerful tool that 
// simulates errors thrown from Statements in our monitored transaction, 
// playing a key role in the test. It functions as a typical mock, 
// but it goes beyond by simulating how a transaction behaves in reality, 
// especially regarding commit time or the object being dropped 
// without an explicit 'commit' call, which should leave the 
// database unmodified (using the roll-back strategy).

// Why is this important? 
// The mocked wrapper has its own transaction, 
// usable for multiple initial calls within the same piece of code. 
// If it interacts with a real transaction, all changes would be persistently 
// written to the database. Considering the various distinct operations 
// in our code, we particularly care about the final stage of the transaction. 
// If an issue prevents reaching the 'commit' call, the staged changes 
// won't be written to the database, as demonstrated in this test.

// Another confirmation of the correct sequence of events is that 
// even without a result prepared for the mocked 'commit' method, 
// the test doesn't panic. If it were committing each transaction 
// separately, we would encounter at least one 
// 'commit' call (resulting in a crash), but we didn't.
```